packages/esm-patient-labs-app/src/test-results/filter/filter-set.tsx

```diff
@@ -18,17 +18,32 @@ interface FilterSetProps {
 }
 
 function filterTreeNode(inputValue, treeNode) {
-  // If the tree node's display value contains the user input, or any of its children's display contains the user input, return true
-  if (
-    treeNode &&
-    (treeNode.display.toLowerCase().includes(inputValue.toLowerCase()) ||
-      (treeNode.subSets && treeNode.subSets.some((child) => filterTreeNode(inputValue, child))))
-  ) {
-    return true;
+  if (!treeNode) {
+    return null;
   }
 
-  // Otherwise, return false
-  return false;
+  const matchesInput = treeNode.display.toLowerCase().includes(inputValue.toLowerCase());
+
+  if (matchesInput) {
+    return treeNode;
+  }
+
+  let filteredSubSets = [];
+
+  if (treeNode.subSets) {
+    filteredSubSets = treeNode.subSets
+      .map((child) => filterTreeNode(inputValue, child))
+      .filter((child) => child !== null);
```

Comment:
This works because Boolean is function that coerces the value to a boolean and `null` gets coerced to `false.

```suggestion
      .filter(Boolean);
```

Reply:
@ibacher are you suggesting a different approach?

Reply:
Yep... I was suggesting you can replace the `.filter(child => child !== null)` with the slightly shorter `.filter(Boolean)`

Reply:
Also in the other similar place.